File: RaftSwift/Sources/RaftCore/Utils/RaftError.swift
/// Errors that can occur in the Raft node.
enum RaftError: Error {
    /// Thrown when a method is called on a node that is not the leader,
    /// but the method requires the node to be the leader.
    case notLeader

    /// Throwsn when a snapshot cannot be created.
    case snapshotCreationError(_ reason: String)
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/types/RaftError.kt
package raft.types

/**
 * Errors that can occur in the Raft node.
 */
sealed class RaftError: Throwable() {
    /**
     * Thrown when a method is called on a node that is not the leader,
     * but the method requires the node to be the leader.
     */
    object NotLeader: RaftError()
}
